src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -145,45 +151,39 @@ def _backchain_one_pass(
                             param_opt_to_nec_pnads[option.parent].append(pnad)
                         break
                 # If we weren't able to find a substitution (i.e, the above
-                # for loop did not break), we need to try specializing each
-                # of our PNADs.
+                # for loop did not break), we need to spawn from the most
+                # general PNAD and make a new PNAD to cover these necessary
+                # add effects.
                 else:
                     nec_pnad_set_changed = True
-                    for pnad in pnads_for_option:
-                        new_pnad = self._try_specializing_pnad(
-                            necessary_add_effects, pnad, segment)
-                        if new_pnad is not None:
-                            assert new_pnad.option_spec == pnad.option_spec
-                            if len(param_opt_to_nec_pnads[option.parent]) > 0:
-                                param_opt_to_nec_pnads[option.parent].remove(
-                                    pnad)
-                            del pnad
-                            break
-                    # If we were unable to specialize any of the PNADs, we need
-                    # to spawn from the most general PNAD and make a new PNAD
-                    # to cover these necessary add effects.
-                    else:
-                        new_pnad = self._try_specializing_pnad(
-                            necessary_add_effects,
-                            param_opt_to_general_pnad[option.parent], segment)
-                        assert new_pnad is not None
-
+                    new_pnad = self._try_specializing_pnad(
```

Comment:
I'm guessing this will go away with #961 but looks okay for now

Reply:
yup - it has since been removed